hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinarySerializer.java

```diff
@@ -655,7 +689,13 @@ private Query writeQueryEdgePrefixCondition(ConditionQuery cq) {
 
             if (key == HugeKeys.OWNER_VERTEX ||
                 key == HugeKeys.OTHER_VERTEX) {
-                buffer.writeId((Id) value);
+                if(this.enablePartition){
+                    String startNode = com.baidu.hugegraph.util.Bytes.toHex(((Id) value).asBytes());
+                    buffer.writeShort(getPartition(startNode, HugeType.EDGE));
+                    buffer.writeId((Id) value);
```

Comment:
can add a method writeVertexId() method and move this logic into writeVertexId()